tests/com/twitter/intellij/pants/util/PantsUtilTest.java

```diff
@@ -57,6 +66,38 @@ public void testFindJdk() {
     assertEquals(twoEntriesSameSdk, getSameJdks(sdkA));
   }
 
+  protected String getPantsExecutablePath() {
+    return PantsUtil.findPantsExecutable(getParentPath()).get().getPath();
+  }
+
+  public void testNonUtf8LocaleSubprocessFails() throws ExecutionException {
+    final GeneralCommandLine command = PantsUtil.defaultCommandLine(getPantsExecutablePath());
+    command.addParameter("options");
+
+    final Map<String, String> localeOverrides = new HashMap<>();
+    final String badLocale = "en_US.US-ASCII";
+    localeOverrides.put("LANG", badLocale);
+    localeOverrides.put("LC_ALL", badLocale);
```

Comment:
nit: you can probably use some of the guava functions to simply this to oneliner.